### PR TITLE
DEV ONLY: Side Navigation // improve stories

### PR DESCRIPTION
**Pull request by joe-s-avaya**
[Link to updated Side Navigation](https://deploy-preview-576--neo-react-library-storybook.netlify.app/?path=/docs/components-side-navigation--docs) | [Link to old (live) Side Navigation stories](https://neo-react-library-storybook.netlify.app/?path=/docs/components-side-navigation--docs)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [ ] ~tagged `@avaya-dux/dux-design` if any visual changes have occurred~
- [x] tagged `@avaya-dux/dux-devs`

As part of the Side Navigation updates, I'm simplifying the stories. 

`@avaya-dux/dux-devs`

- compare old and new stories
- note that the new stories have controls, but they only control how the story looks, no actual component prop changes

## Summary by CodeRabbit

- **New Features**
	- Enhanced configurability of the `SideNavigation` component with new properties for better customization in the Storybook environment.
	- Introduced a new `MultipleLevels` story for demonstrating complex navigation structures.

- **Bug Fixes**
	- Updated the `href` prop in the `TopLinkItem` component to ensure it defaults to an empty string, improving robustness.

- **Tests**
	- Updated test cases for `TopLinkItem` to include the new `href` prop and ensure correct navigation behavior.

- **Chores**
	- Streamlined the test suite by removing outdated stories and focusing on relevant components.

### CODE REVIEW

File: src/components/SideNavigation/SideNavigationTypes.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import type { IconNamesType } from "utils";
 
 export interface SideNavigationTopLinkItemProps {
 	label: string;
-	href: string;
+	href?: string;
```
Comment (joe-s-avaya) [inline src/components/SideNavigation/SideNavigationTypes.tsx:7]: IMPORTANT: this allows for disabled Top Link Items (and turns them into buttons)

**Review by yangAtSpoken — COMMENTED** [review]
Screenshot 2024-10-28 at 5 54 45 PM

Multi level rtl does not look right. Not sure if it is in the scope of this PR.

**Comment by joe-s-avaya** [comment]
> Multi level rtl does not look right. Not sure if it is in the scope of this PR.

@yangAtSpoken good call. I just checked the [live site](https://neo-react-library-storybook.netlify.app/?path=/story/components-side-navigation--category-groups) and it looks the same. So outside of the scope of this PR. But I will make sure to resolve that as I make updates to the Side Nav component. 

Screenshot 2024-10-29 at 09 13 44

**Review by yangAtSpoken — APPROVED** [review]

File: src/components/SideNavigation/SideNavigation.stories.tsx [hunk 2]
```diff
@@ -1,188 +1,212 @@
...
-				</SideNavigation.LinkItem>
-			</SideNavigation.NavCategory>
-		</SideNavigation>
-	),
-};
+	argTypes: {
+		dir: {
+			control: "radio",
+			options: ["ltr", "rtl"],
+			description: "These are props for the story, not the component.",
```
Comment (mrazian85) [inline src/components/SideNavigation/SideNavigation.stories.tsx:38]: 👍

**Review by mrazian85 — APPROVED** [review]
Other than what Yang pointed out about Multiple Levels not looking right in RTL, it LGTM 👍

**Pull request merged**